### ISSUE TITLE
Failed to exec command: sudo -E /bin/bash -c "/usr/local/bin/kubeadm init --config=/etc/kubernetes/kubeadm-config.yaml --ignore-preflight-errors=FileExisting-crictl,ImagePull"  your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version. To see the stack trace of this error execute with --v=5 or higher: Process exited with status 1

### ISSUE THREAD

**Issue by wuruiwm** [kubesphere/kubekey#1883]
### What is version of KubeKey has the issue?

v3.1.0-alpha.2

### What is your os environment?

debian11

### KubeKey config file

```yaml
apiVersion: kubekey.kubesphere.io/v1alpha2
kind: Cluster
metadata:
  name: k8s
spec:
  hosts:
  - {name: master-01, address: 10.10.10.104, internalAddress: 10.10.10.104, user: root, password: "123456", port: 2222}
  roleGroups:
    etcd:
    - master-01
    control-plane: 
    - master-01
    worker:
    - master-01
  controlPlaneEndpoint:
    domain: k8s
    address: ""
    port: 6443
  kubernetes:
    version: v1.27.2
    clusterName: k8s
    autoRenewCerts: true
    containerManager: containerd
  etcd:
    type: kubekey
  network:
    plugin: calico
    kubePodsCIDR: 10.233.64.0/18
    kubeServiceCIDR: 10.233.0.0/18
    multusCNI:
      enabled: false
  registry:
    privateRegistry: ""
    namespaceOverride: ""
    registryMirrors: ["https://tecje8ji.mirror.aliyuncs.com","https://hub-mirror.c.163.com","https://docker.mirrors.ustc.edu.cn","https://a5nu9ynx.mirror.aliyuncs.com"]
    insecureRegistries: []
  addons: []
```

### A clear and concise description of what happend.

```
root@localhost:~# ./kk create cluster -f config-sample.yaml


 _   __      _          _   __           
| | / /     | |        | | / /           
| |/ / _   _| |__   ___| |/ /  ___ _   _ 
|    \| | | | '_ \ / _ \    \ / _ \ | | |
| |\  \ |_| | |_) |  __/ |\  \  __/ |_| |
\_| \_/\__,_|_.__/ \___\_| \_/\___|\__, |
                                    __/ |
                                   |___/

15:40:29 CST [GreetingsModule] Greetings
15:40:29 CST message: [master-01]
sudo: unable to resolve host master-01: Name or service not known
Greetings, KubeKey!
15:40:29 CST success: [master-01]
15:40:29 CST [NodePreCheckModule] A pre-check on nodes
15:40:29 CST success: [master-01]
15:40:29 CST [ConfirmModule] Display confirmation form
+-----------+------+------+---------+----------+-------+-------+---------+-----------+--------+--------+------------+------------+-------------+------------------+--------------+
| name      | sudo | curl | openssl | ebtables | socat | ipset | ipvsadm | conntrack | chrony | docker | containerd | nfs client | ceph client | glusterfs client | time         |
+-----------+------+------+---------+----------+-------+-------+---------+-----------+--------+--------+------------+------------+-------------+------------------+--------------+
| master-01 | y    | y    | y       | y        | y     |       |         | y         |        |        |            |            |             |                  | CST 15:40:29 |
+-----------+------+------+---------+----------+-------+-------+---------+-----------+--------+--------+------------+------------+-------------+------------------+--------------+

This is a simple check of your environment.
Before installation, ensure that your machines meet all requirements specified at
https://github.com/kubesphere/kubekey#requirements-and-recommendations

Continue this installation? [yes/no]: yes
15:40:32 CST success: [LocalHost]
15:40:32 CST [NodeBinariesModule] Download installation binaries
15:40:32 CST message: [localhost]
downloading amd64 kubeadm v1.27.2 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 45.9M  100 45.9M    0     0  1030k      0  0:00:45  0:00:45 --:--:-- 1192k
15:41:18 CST message: [localhost]
downloading amd64 kubelet v1.27.2 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  101M  100  101M    0     0  1019k      0  0:01:41  0:01:41 --:--:-- 1014k
15:43:00 CST message: [localhost]
downloading amd64 kubectl v1.27.2 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 46.9M  100 46.9M    0     0  1030k      0  0:00:46  0:00:46 --:--:-- 1219k
15:43:47 CST message: [localhost]
downloading amd64 helm v3.9.0 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 44.0M  100 44.0M    0     0  1021k      0  0:00:44  0:00:44 --:--:-- 1066k
15:44:31 CST message: [localhost]
downloading amd64 kubecni v1.2.0 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 38.6M  100 38.6M    0     0  1031k      0  0:00:38  0:00:38 --:--:-- 1200k
15:45:10 CST message: [localhost]
downloading amd64 crictl v1.24.0 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.8M  100 13.8M    0     0  1077k      0  0:00:13  0:00:13 --:--:-- 1207k
15:45:23 CST message: [localhost]
downloading amd64 etcd v3.4.13 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 16.5M  100 16.5M    0     0  1027k      0  0:00:16  0:00:16 --:--:-- 1082k
15:45:40 CST message: [localhost]
downloading amd64 containerd 1.6.4 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 42.3M  100 42.3M    0     0  1031k      0  0:00:42  0:00:42 --:--:-- 1199k
15:46:22 CST message: [localhost]
downloading amd64 runc v1.1.1 ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9194k  100 9194k    0     0  1105k      0  0:00:08  0:00:08 --:--:-- 1202k
15:46:30 CST success: [LocalHost]
15:46:30 CST [ConfigureOSModule] Get OS release
15:46:30 CST success: [master-01]
15:46:30 CST [ConfigureOSModule] Prepare to init OS
15:46:31 CST success: [master-01]
15:46:31 CST [ConfigureOSModule] Generate init os script
15:46:31 CST success: [master-01]
15:46:31 CST [ConfigureOSModule] Exec init os script
15:46:31 CST stdout: [master-01]
net.ipv4.ip_forward = 1
net.bridge.bridge-nf-call-arptables = 1
net.bridge.bridge-nf-call-ip6tables = 1
net.bridge.bridge-nf-call-iptables = 1
net.ipv4.ip_local_reserved_ports = 30000-32767
vm.max_map_count = 262144
vm.swappiness = 1
fs.inotify.max_user_instances = 524288
kernel.pid_max = 65535
15:46:31 CST success: [master-01]
15:46:31 CST [ConfigureOSModule] configure the ntp server for each node
15:46:31 CST skipped: [master-01]
15:46:31 CST [KubernetesStatusModule] Get kubernetes cluster status
15:46:31 CST success: [master-01]
15:46:31 CST [InstallContainerModule] Sync containerd binaries
15:46:32 CST success: [master-01]
15:46:32 CST [InstallContainerModule] Sync crictl binaries
15:46:33 CST success: [master-01]
15:46:33 CST [InstallContainerModule] Generate containerd service
15:46:33 CST success: [master-01]
15:46:33 CST [InstallContainerModule] Generate containerd config
15:46:33 CST success: [master-01]
15:46:33 CST [InstallContainerModule] Generate crictl config
15:46:33 CST success: [master-01]
15:46:33 CST [InstallContainerModule] Enable containerd
15:46:33 CST success: [master-01]
15:46:33 CST [PullModule] Start to pull images on all nodes
15:46:33 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/pause:3.8
15:46:35 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/kube-apiserver:v1.27.2
15:46:54 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/kube-controller-manager:v1.27.2
15:47:10 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/kube-scheduler:v1.27.2
15:47:20 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/kube-proxy:v1.27.2
15:47:30 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/coredns:1.9.3
15:47:38 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/k8s-dns-node-cache:1.15.12
15:47:51 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/kube-controllers:v3.23.2
15:48:06 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/cni:v3.23.2
15:48:48 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/node:v3.23.2
15:49:28 CST message: [master-01]
downloading image: registry.cn-beijing.aliyuncs.com/kubesphereio/pod2daemon-flexvol:v3.23.2
15:49:32 CST success: [master-01]
15:49:32 CST [ETCDPreCheckModule] Get etcd status
15:49:32 CST success: [master-01]
15:49:32 CST [CertsModule] Fetch etcd certs
15:49:32 CST success: [master-01]
15:49:32 CST [CertsModule] Generate etcd Certs
[certs] Generating "ca" certificate and key
[certs] admin-master-01 serving cert is signed for DNS names [etcd etcd.kube-system etcd.kube-system.svc etcd.kube-system.svc.cluster.local k8s localhost master-01] and IPs [127.0.0.1 ::1 10.10.10.107]
[certs] member-master-01 serving cert is signed for DNS names [etcd etcd.kube-system etcd.kube-system.svc etcd.kube-system.svc.cluster.local k8s localhost master-01] and IPs [127.0.0.1 ::1 10.10.10.107]
[certs] node-master-01 serving cert is signed for DNS names [etcd etcd.kube-system etcd.kube-system.svc etcd.kube-system.svc.cluster.local k8s localhost master-01] and IPs [127.0.0.1 ::1 10.10.10.107]
15:49:33 CST success: [LocalHost]
15:49:33 CST [CertsModule] Synchronize certs file
15:49:33 CST success: [master-01]
15:49:33 CST [CertsModule] Synchronize certs file to master
15:49:33 CST skipped: [master-01]
15:49:33 CST [InstallETCDBinaryModule] Install etcd using binary
15:49:34 CST success: [master-01]
15:49:34 CST [InstallETCDBinaryModule] Generate etcd service
15:49:34 CST success: [master-01]
15:49:34 CST [InstallETCDBinaryModule] Generate access address
15:49:34 CST success: [master-01]
15:49:34 CST [ETCDConfigureModule] Health check on exist etcd
15:49:34 CST skipped: [master-01]
15:49:34 CST [ETCDConfigureModule] Generate etcd.env config on new etcd
15:49:34 CST success: [master-01]
15:49:34 CST [ETCDConfigureModule] Refresh etcd.env config on all etcd
15:49:34 CST success: [master-01]
15:49:34 CST [ETCDConfigureModule] Restart etcd
15:49:39 CST stdout: [master-01]
Created symlink /etc/systemd/system/multi-user.target.wants/etcd.service → /etc/systemd/system/etcd.service.
15:49:39 CST success: [master-01]
15:49:39 CST [ETCDConfigureModule] Health check on all etcd
15:49:39 CST success: [master-01]
15:49:39 CST [ETCDConfigureModule] Refresh etcd.env config to exist mode on all etcd
15:49:39 CST success: [master-01]
15:49:39 CST [ETCDConfigureModule] Health check on all etcd
15:49:39 CST success: [master-01]
15:49:39 CST [ETCDBackupModule] Backup etcd data regularly
15:49:39 CST success: [master-01]
15:49:39 CST [ETCDBackupModule] Generate backup ETCD service
15:49:39 CST success: [master-01]
15:49:39 CST [ETCDBackupModule] Generate backup ETCD timer
15:49:39 CST success: [master-01]
15:49:39 CST [ETCDBackupModule] Enable backup etcd service
15:49:39 CST success: [master-01]
15:49:39 CST [InstallKubeBinariesModule] Synchronize kubernetes binaries
15:49:43 CST success: [master-01]
15:49:43 CST [InstallKubeBinariesModule] Change kubelet mode
15:49:43 CST success: [master-01]
15:49:43 CST [InstallKubeBinariesModule] Generate kubelet service
15:49:44 CST success: [master-01]
15:49:44 CST [InstallKubeBinariesModule] Enable kubelet service
15:49:44 CST success: [master-01]
15:49:44 CST [InstallKubeBinariesModule] Generate kubelet env
15:49:44 CST success: [master-01]
15:49:44 CST [InitKubernetesModule] Generate kubeadm config
15:49:44 CST success: [master-01]
15:49:44 CST [InitKubernetesModule] Init cluster using kubeadm
15:49:44 CST stdout: [master-01]
your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
To see the stack trace of this error execute with --v=5 or higher
15:49:44 CST stdout: [master-01]
[preflight] Running pre-flight checks
W0621 15:49:44.611616    2805 removeetcdmember.go:106] [reset] No kubeadm config, using etcd pod spec to get data directory
[reset] Deleted contents of the etcd data directory: /var/lib/etcd
[reset] Stopping the kubelet service
[reset] Unmounting mounted directories in "/var/lib/kubelet"
W0621 15:49:44.617449    2805 cleanupnode.go:134] [reset] Failed to evaluate the "/var/lib/kubelet" directory. Skipping its unmount and cleanup: lstat /var/lib/kubelet: no such file or directory
[reset] Deleting contents of directories: [/etc/kubernetes/manifests /etc/kubernetes/pki]
[reset] Deleting files: [/etc/kubernetes/admin.conf /etc/kubernetes/kubelet.conf /etc/kubernetes/bootstrap-kubelet.conf /etc/kubernetes/controller-manager.conf /etc/kubernetes/scheduler.conf]

The reset process does not clean CNI configuration. To do so, you must remove /etc/cni/net.d

The reset process does not reset or clean up iptables rules or IPVS tables.
If you wish to reset iptables, you must do so manually by using the "iptables" command.

If your cluster was setup to utilize IPVS, run ipvsadm --clear (or similar)
to reset your system's IPVS tables.

The reset process does not clean your kubeconfig files and you must remove them manually.
Please, check the contents of the $HOME/.kube/config file.
15:49:44 CST message: [master-01]
init kubernetes cluster failed: Failed to exec command: sudo -E /bin/bash -c "/usr/local/bin/kubeadm init --config=/etc/kubernetes/kubeadm-config.yaml --ignore-preflight-errors=FileExisting-crictl,ImagePull" 
your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
To see the stack trace of this error execute with --v=5 or higher: Process exited with status 1
15:49:44 CST retry: [master-01]
15:49:49 CST stdout: [master-01]
your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
To see the stack trace of this error execute with --v=5 or higher
15:49:49 CST stdout: [master-01]
[preflight] Running pre-flight checks
W0621 15:49:49.707446    2821 removeetcdmember.go:106] [reset] No kubeadm config, using etcd pod spec to get data directory
[reset] Stopping the kubelet service
[reset] Unmounting mounted directories in "/var/lib/kubelet"
W0621 15:49:49.711371    2821 cleanupnode.go:134] [reset] Failed to evaluate the "/var/lib/kubelet" directory. Skipping its unmount and cleanup: lstat /var/lib/kubelet: no such file or directory
[reset] Deleting contents of directories: [/etc/kubernetes/manifests /etc/kubernetes/pki]
[reset] Deleting files: [/etc/kubernetes/admin.conf /etc/kubernetes/kubelet.conf /etc/kubernetes/bootstrap-kubelet.conf /etc/kubernetes/controller-manager.conf /etc/kubernetes/scheduler.conf]

The reset process does not clean CNI configuration. To do so, you must remove /etc/cni/net.d

The reset process does not reset or clean up iptables rules or IPVS tables.
If you wish to reset iptables, you must do so manually by using the "iptables" command.

If your cluster was setup to utilize IPVS, run ipvsadm --clear (or similar)
to reset your system's IPVS tables.

The reset process does not clean your kubeconfig files and you must remove them manually.
Please, check the contents of the $HOME/.kube/config file.
15:49:49 CST message: [master-01]
init kubernetes cluster failed: Failed to exec command: sudo -E /bin/bash -c "/usr/local/bin/kubeadm init --config=/etc/kubernetes/kubeadm-config.yaml --ignore-preflight-errors=FileExisting-crictl,ImagePull" 
your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
To see the stack trace of this error execute with --v=5 or higher: Process exited with status 1
15:49:49 CST retry: [master-01]
15:49:54 CST stdout: [master-01]
your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
To see the stack trace of this error execute with --v=5 or higher
15:49:54 CST stdout: [master-01]
[preflight] Running pre-flight checks
W0621 15:49:54.859310    2839 removeetcdmember.go:106] [reset] No kubeadm config, using etcd pod spec to get data directory
[reset] Stopping the kubelet service
[reset] Unmounting mounted directories in "/var/lib/kubelet"
W0621 15:49:54.863091    2839 cleanupnode.go:134] [reset] Failed to evaluate the "/var/lib/kubelet" directory. Skipping its unmount and cleanup: lstat /var/lib/kubelet: no such file or directory
[reset] Deleting contents of directories: [/etc/kubernetes/manifests /etc/kubernetes/pki]
[reset] Deleting files: [/etc/kubernetes/admin.conf /etc/kubernetes/kubelet.conf /etc/kubernetes/bootstrap-kubelet.conf /etc/kubernetes/controller-manager.conf /etc/kubernetes/scheduler.conf]

The reset process does not clean CNI configuration. To do so, you must remove /etc/cni/net.d

The reset process does not reset or clean up iptables rules or IPVS tables.
If you wish to reset iptables, you must do so manually by using the "iptables" command.

If your cluster was setup to utilize IPVS, run ipvsadm --clear (or similar)
to reset your system's IPVS tables.

The reset process does not clean your kubeconfig files and you must remove them manually.
Please, check the contents of the $HOME/.kube/config file.
15:49:54 CST message: [master-01]
init kubernetes cluster failed: Failed to exec command: sudo -E /bin/bash -c "/usr/local/bin/kubeadm init --config=/etc/kubernetes/kubeadm-config.yaml --ignore-preflight-errors=FileExisting-crictl,ImagePull" 
your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
To see the stack trace of this error execute with --v=5 or higher: Process exited with status 1
15:49:54 CST failed: [master-01]
error: Pipeline[CreateClusterPipeline] execute failed: Module[InitKubernetesModule] exec failed: 
failed: [master-01] [KubeadmInit] exec failed after 3 retries: init kubernetes cluster failed: Failed to exec command: sudo -E /bin/bash -c "/usr/local/bin/kubeadm init --config=/etc/kubernetes/kubeadm-config.yaml --ignore-preflight-errors=FileExisting-crictl,ImagePull" 
your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
To see the stack trace of this error execute with --v=5 or higher: Process exited with status 1

```

**Comment by littleBlackHouse** [comment]
It's possible that the current version of KubeKey doesn't support installing Kubernetes version 1.27.2. You can use the command `kk version --show-supported-k8s` to check the supported versions.

**Comment by wuruiwm** [comment]
支持的
```
root@localhost:~# ./kk version --show-supported-k8s
v1.19.0
v1.19.8
v1.19.9
v1.19.15
v1.20.4
v1.20.6
v1.20.10
v1.21.0
v1.21.1
v1.21.2
v1.21.3
v1.21.4
v1.21.5
v1.21.6
v1.21.7
v1.21.8
v1.21.9
v1.21.10
v1.21.11
v1.21.12
v1.21.13
v1.21.14
v1.22.0
v1.22.1
v1.22.2
v1.22.3
v1.22.4
v1.22.5
v1.22.6
v1.22.7
v1.22.8
v1.22.9
v1.22.10
v1.22.11
v1.22.12
v1.22.13
v1.22.14
v1.22.15
v1.22.16
v1.22.17
v1.23.0
v1.23.1
v1.23.2
v1.23.3
v1.23.4
v1.23.5
v1.23.6
v1.23.7
v1.23.8
v1.23.9
v1.23.10
v1.23.11
v1.23.12
v1.23.13
v1.23.14
v1.23.15
v1.23.16
v1.23.17
v1.24.0
v1.24.1
v1.24.2
v1.24.3
v1.24.4
v1.24.5
v1.24.6
v1.24.7
v1.24.8
v1.24.9
v1.24.10
v1.24.11
v1.24.12
v1.24.13
v1.24.14
v1.25.0
v1.25.1
v1.25.2
v1.25.3
v1.25.4
v1.25.5
v1.25.6
v1.25.7
v1.25.8
v1.25.9
v1.25.10
v1.26.0
v1.26.1
v1.26.2
v1.26.3
v1.26.4
v1.26.5
v1.27.0
v1.27.1
v1.27.2

```

**Comment by littleBlackHouse** [comment]
@wuruiwm Thank you for your issue.
refer to https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta3/ 
<img width="735" alt="截屏2023-06-25 09 43 46" src="https://github.com/kubesphere/kubekey/assets/54946465/bad7c4b3-9318-41ce-ae24-cdd216e99a07">
We will create a new PR to fix this issue, and if you have the time, you are also welcome to contribute and fix it.
you can temporarily use version 3.0.7.

**Comment by yzcyayaya** [comment]
> 

I have also encountered this problem, can you assign it to me? I'm interested in contributing to the community.

```
init kubernetes cluster failed: Failed to exec command: sudo -E /bin/bash -c "/usr/local/bin/kubeadm init --config=/etc/kubernetes/kubeadm-config.yaml --ignore-preflight-errors=FileExisting-crictl,ImagePull"
your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
To see the stack trace of this error execute with --v=5 or higher: Process exited with status 1
```
And I also found that the etcd health check is no longer applicable  
> ==> So far I have implemented the installation by code removing the health check
```
etcd health check failed: Failed to exec command: sudo -E /bin/bash -c "export ETCDCTL_API=2;export ETCDCTL_CERT_FILE='/etc/ssl/etcd/ssl/admin-master.pem';export ETCDCTL_KEY_FILE='/etc/ssl/etcd/ssl/admin-master-key.pem';export ETCDCTL_CA_FILE='/etc/ssl/etcd/ssl/ca.pem';/usr/local/bin/etcdctl --endpoints=https://192.168.1.4:2379 cluster-health | grep -q 'cluster is healthy'" 
Error:  client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 192.168.1.4:2379: connect: connection refused
```
/cc @littleBlackHouse

**Comment by littleBlackHouse** [comment]
> I have also encountered this problem, can you assign it to me? I'm interested in contributing to the community.
> 
> ```
> init kubernetes cluster failed: Failed to exec command: sudo -E /bin/bash -c "/usr/local/bin/kubeadm init --config=/etc/kubernetes/kubeadm-config.yaml --ignore-preflight-errors=FileExisting-crictl,ImagePull"
> your configuration file uses an old API spec: "kubeadm.k8s.io/v1beta2". Please use kubeadm v1.22 instead and run 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.
> To see the stack trace of this error execute with --v=5 or higher: Process exited with status 1
> ```
> 
> And I also found that the etcd health check is no longer applicable
> 
> > ==> So far I have implemented the installation by code removing the health check
> 
> ```
> etcd health check failed: Failed to exec command: sudo -E /bin/bash -c "export ETCDCTL_API=2;export ETCDCTL_CERT_FILE='/etc/ssl/etcd/ssl/admin-master.pem';export ETCDCTL_KEY_FILE='/etc/ssl/etcd/ssl/admin-master-key.pem';export ETCDCTL_CA_FILE='/etc/ssl/etcd/ssl/ca.pem';/usr/local/bin/etcdctl --endpoints=https://192.168.1.4:2379 cluster-health | grep -q 'cluster is healthy'" 
> Error:  client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 192.168.1.4:2379: connect: connection refused
> ```
> 
> /cc @littleBlackHouse

Of course! Welcome!👏👏

**Comment by yzcyayaya** [comment]
sorry, I didn't fix this,But I found another problem
kubelet status failed
run
```shell
journalctl -xeu kubelet
```
logs
```shell
Aug 05 17:10:19 master kubelet[56018]: Flag --cgroup-driver has been deprecated, This parameter should be set via the config file specified by >
Aug 05 17:10:19 master kubelet[56018]: Flag --container-runtime-endpoint has been deprecated, This parameter should be set via the config file >
Aug 05 17:10:19 master kubelet[56018]: Flag --pod-infra-container-image has been deprecated, will be removed in a future release. Image garbage>
Aug 05 17:10:19 master kubelet[56018]: I0805 17:10:19.771137   56018 server.go:199] "--pod-infra-container-image will not be pruned by the imag>
Aug 05 17:10:19 master kubelet[56018]: W0805 17:10:19.772101   56018 feature_gate.go:241] Setting GA feature gate CSIStorageCapacity=true. It w>
Aug 05 17:10:19 master kubelet[56018]: E0805 17:10:19.772133   56018 run.go:74] "command failed" err="failed to set feature gates from initial >
Aug 05 17:10:19 master systemd[1]: kubelet.service: Main process exited, code=exited, status=1/FAILURE
```

ExpandCSIVolumes  at v1.27 already remove
at page ctrl + f  search > ExpandCSIVolumes
 [v1.27](https://kubernetes.io/zh-cn/blog/2023/03/17/upcoming-changes-in-kubernetes-v1-27/)

/cc @littleBlackHouse

**Comment by Taurus-Le** [comment]
Just tried with kubekey 3.0.10 and hit this problem again.

**Comment by zwz9599** [comment]
> Just tried with kubekey 3.0.10 and hit this problem again.

Still have the same problem with this version

**Comment by dommgifer** [comment]
3.0.10 still have the same problem

**Comment by littleBlackHouse** [comment]
Resolved in version v3.1.0-alpha.5.
PR is: https://github.com/kubesphere/kubekey/pull/2000
![image](https://github.com/kubesphere/kubekey/assets/54946465/2c17f797-1800-43e5-b67a-0c06aa1e6448)
